capsules/extra/src/nonvolatile_storage_driver.rs

```diff
@@ -538,9 +1283,10 @@ impl SyscallDriver for NonvolatileStorage<'_> {
     /// ### `command_num`
     ///
     /// - `0`: Return Ok(()) if this driver is included on the platform.
-    /// - `1`: Return the number of bytes available to userspace.
+    /// - `1`: Return the number of bytes available to each app.
     /// - `2`: Start a read from the nonvolatile storage.
     /// - `3`: Start a write to the nonvolatile_storage.
+    /// - `4`: Initialize an app's nonvolatile_storage.
```

Comment:
How does this operation work? What happens if it is called more than once? What are the requirements for apps?

Reply:
Init will either find an app's region that's already in storage or it will allocate a new one.

Apps must call this before any of the other syscalls (read/write/get_num_bytes) and wait for the init callback to fire.

There's also the requirements for setting up stuff in the `main.rs` like setting up `AppID` assignment and giving storage permissions but that's more of the responsibility of whoever is flashing the board.